### PR TITLE
ML notebook formatting and text updates

### PR DESCRIPTION
**Pull request by rszper**
* Updated some formatting, such as headers and how the license appears on devsite
* Copy edited the text

R: @damccorm 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb [hunk 1]
```diff
@@ -38,29 +38,23 @@
...
-        "This notebook demonstrates the use of the Apache Beam DataFrames API to perform common data exploration as well as the preprocessing steps that are necessary to prepare your dataset for machine learning model training and inference. These steps include the following:  \n",
+        "This example demonstrates the use of the Apache Beam DataFrames API to perform common data exploration as well as the preprocessing steps that are necessary to prepare your dataset for machine learning model training and inference. This example includes the following steps:  \n",
         "\n",
         "*   Removing unwanted columns.\n",
         "*   One-hot encoding categorical columns.\n",
         "*   Normalizing numerical columns.\n",
         "\n",
-        "\n"
+        "In this example, the first section demonstrates how to build and execute a pipeline locally using the interactive runner.\n",
+        "The second section uses a distributed runner to demonstrate how to run the pipeline on the full dataset.\n",
```
Comment (damccorm) [inline examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb:57]:
```suggestion
        "The second section uses a distributed runner to demonstrate how to run the pipeline on the full dataset.\n"
```

Looks like this notebook shows up as invalid - https://github.com/apache/beam/blob/d29825911841635ba03c7d2d80943d0326d149de/examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb - this may be the problem?

Comment (damccorm) [reply examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb:57]: Until we have tests, we should probably hand validate these notebooks before merging every time

File: examples/notebooks/beam-ml/run_inference_multi_model.ipynb [hunk 2]
```diff
@@ -520,7 +520,6 @@
         "\n",
         "  def process(self, element):\n",
         "    image_url, image = element \n",
-        "    # Update this step when this ticket is resolved: https://github.com/apache/beam/issues/21863\n",
```
Comment (damccorm) [inline examples/notebooks/beam-ml/run_inference_multi_model.ipynb:523]: Hm, is there a better way to communicate these kinds of workarounds are temporary?

Comment (damccorm) [reply examples/notebooks/beam-ml/run_inference_multi_model.ipynb:523]: If not, we should at least comment on the issue itself with a note to update the notebook when it is fixed

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb [hunk 3]
```diff
@@ -84,6 +84,13 @@
       "metadata": {
         "id": "loxD-rOVchRn"
       },
+      "outputs": [],
+      "source": [
+        "!pip install apache_beam[gcp,dataframe] --quiet"
+      ],
+      "metadata": {
+        "id": "loxD-rOVchRn"
+      },
```
Comment (damccorm) [inline examples/notebooks/beam-ml/run_inference_pytorch.ipynb:93]: Was this change intentional? It looks like it no-ops

File: examples/notebooks/beam-ml/run_inference_pytorch_tensorflow_sklearn.ipynb [hunk 4]
```diff
@@ -41,25 +36,23 @@
...
     {
       "cell_type": "markdown",
       "metadata": {
         "id": "JjAt1GesQ9sg"
       },
       "source": [
-        "Starting with Apache Beam 2.40.0, you can use Apache Beam with the RunInference API to use machine learning (ML) models for local and remote inference with batch and streaming pipelines.\n",
-        "The RunInference API leverages Apache Beam concepts, such as the BatchElements transform and the Shared class, to support models in your pipelines that create transforms optimized for machine learning inferences.\n",
+        "# Use RunInference in Apache Beam\n",
+        "You can use Apache Beam versions 2.40.0 and later with the [RunInference API](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.base.html#apache_beam.ml.inference.base.RunInference) to use machine learning (ML) models for local and remote inference with batch and streaming pipelines.\n",
```
Comment (damccorm) [inline examples/notebooks/beam-ml/run_inference_pytorch_tensorflow_sklearn.ipynb:54]: "You can use Apache Beam versions 2.40.0 and later with the RunInference API to use machine learning (ML) models for local and remote inference with batch and streaming pipelines." - this reads a little awkwardly to me, specifically "you can use X to use Y"

Maybe "You can use Apache Beam versions 2.40.0 and later with the RunInference API for local and remote inference with batch and streaming pipelines" instead? The "to use machine learning (ML) models" bit is probably implied

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb (hunk 3 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_pytorch.ipynb:93]: I think this is a change that Anand made, and for some reason it's showing up in this PR: https://github.com/apache/beam/pull/24336/files#diff-d74554e3521bce6230681b255694c89efcceae49a13c2baccb9b91d983b1d6fc

I can try to get rid of it again, but it keeps showing up.

File: examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb (hunk 1 above)
Comment (damccorm) [reply examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb]: That fixed it - https://github.com/rszper/beam/blob/rszper-ML-notebooks/examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb (hunk 3 above)
Comment (damccorm) [reply examples/notebooks/beam-ml/run_inference_pytorch.ipynb:93]: The change itself is good, but source/metadata/outputs are duplicated in your version right now

File: examples/notebooks/beam-ml/run_inference_multi_model.ipynb (hunk 2 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_multi_model.ipynb:523]: I think this is the difference between GitHub content and Devsite content. We can definitely note this as a workaround, but we need to be careful about the phrasing. I'll take a look at the Beam issue and see what I can come up with.

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb (hunk 3 above)
Comment (damccorm) [reply examples/notebooks/beam-ml/run_inference_pytorch.ipynb:93]: Probably just a weird merge in of master, you should just be able to delete these lines though

File: examples/notebooks/beam-ml/run_inference_multi_model.ipynb (hunk 2 above)
Comment (damccorm) [reply examples/notebooks/beam-ml/run_inference_multi_model.ipynb:523]: Ok, thanks!

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb (hunk 3 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_pytorch.ipynb:93]: I think this was a bad merge after a conflict. I'll revert it.

File: examples/notebooks/beam-ml/run_inference_multi_model.ipynb (hunk 2 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_multi_model.ipynb:523]: Replaced with:

# The following lines provide a workaround to turn off BatchElements.

In all four spots.

File: examples/notebooks/beam-ml/run_inference_pytorch.ipynb (hunk 3 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_pytorch.ipynb]: I think I successfully got rid of this.

File: examples/notebooks/beam-ml/run_inference_pytorch_tensorflow_sklearn.ipynb (hunk 4 above)
Comment (rszper) [reply examples/notebooks/beam-ml/run_inference_pytorch_tensorflow_sklearn.ipynb]: Updated, thanks!

**Review by damccorm — APPROVED** [review]
LGTM, thanks!
